src/cache.c

```diff
@@ -144,35 +145,32 @@ cache_fetch(Cache *cache, CacheQuery *query)
 	HASHACTION	action = cache->create_entry == NULL ? HASH_FIND : HASH_ENTER;
 
 	if (cache->htab == NULL)
-	{
-		elog(ERROR, "Hash %s not initialized", cache->name);
-	}
+		elog(ERROR, "hash %s is not initialized", cache->name);
 
+	if (query->enteronly)
```

Comment:
Shouldn't this be part of the action definition above? ie line 145 should be `HASHACTION	action = cache->create_entry == NULL && !query->enteronly ? HASH_FIND : HASH_ENTER;`